src/kolibri_gnome/kolibri_context.py

```diff
@@ -211,6 +217,13 @@ def parse_x_kolibri_app_url_tuple(self, url_tuple: SplitResult) -> str:
         """
         return url_tuple._replace(scheme="", netloc="").geturl()
 
+    def __webkit_web_context_on_download_started(
+        self, webkit_web_context: WebKit2.WebContext, download: WebKit2.Download
+    ):
+        url = download.get_request().get_uri()
+
+        self.emit("download-started", download)
```

Comment:
This class captures a signal just to emit a new one with same name. But it seems that's how GTK works.